### PR TITLE
*: clear and refine some code for RPC methods

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

remove some useless code

### What is changed and how it works?

- remove some commands permanently
    - remove `migrate-relay`: never used it even in v1.0
    - remove `switch-relay-master`: will support it when implementing update-source online and refine on relay
    - remove `update-relay`: will support it when implementing update-source online and refine on relay
    - remove `update-master-config`: register DM-worker dynamically in v2.0
- remove some methods for RPC methods
    - remove `StartSubTask` in DM-worker: watch task operation via etcd
    - remove `OperateSubTask` in DM-worker: watch task operation via etcd
    - remove `UpdateSubTask ` in DM-worker: will re-implement it via watch etcd
    - remove `QueryWorkerConfig` in DM-worker: no need in v2.0
    - remove `OperateRelay` in DM-worker: will re-implement it via watch etcd
- remove config items
    - remove `deploy` in DM-master: register DM-worker dynamically in v2.0

refine some test cases.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: dm/master/bootstrap_test.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ func (t *testMaster) TestCollectSourceConfigFilesV1Import(c *C) {
 	cfgs, err = s.collectSourceConfigFilesV1Import(tctx)
 	c.Assert(terror.ErrConfigYamlTransform.Equal(err), IsTrue)
 	c.Assert(cfgs, HasLen, 0)
+	clearEtcdEnv(c)
```
Comment (lance6716) [inline dm/master/bootstrap_test.go:101]: https://godoc.org/github.com/stretchr/testify/suite#TearDownTestSuite

File: dm/master/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -99,7 +99,7 @@ func (t *testScheduler) testSchedulerProgress(c *C, restart int) {
 		workerInfo2  = ha.NewWorkerInfo(workerName2, workerAddr2)
 		sourceCfg1   config.SourceConfig
 		subtaskCfg1  config.SubTaskConfig
-		keepAliveTTL = int64(2) // NOTE: this should be >= minLeaseTTL, in second.
+		keepAliveTTL = int64(5) // NOTE: this should be >= minLeaseTTL, in second.
```
Comment (lance6716) [inline dm/master/scheduler/scheduler_test.go:102]: I'll revert https://github.com/pingcap/dm/pull/1046/commits/06cad5321f3073e38372650512fac54580c223cf

**Review by GMHDBJD — COMMENTED** [review]
LGTM

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
LGTM

**Review by lance6716 — APPROVED** [review]

**Comment by csuzhangxc** [comment]
/run-all-tests

`java.nio.channels.ClosedChannelException`

**Pull request merged**